### ISSUE TITLE
[FEATURE]  Automatically play next book in a series

### ISSUE THREAD

**Issue by kgraefe** [LeoKlaus/plappa#124]
I have a kids series with hundreds of short episodes (~15min). We mostly listen to them in the car where I cannot manually select a new episode every 15min. It would be very desirable if Plappa would automatically continue with the next episode if one is finished.

**Comment by LeoKlaus** [comment]
Hey Konrad,

I've added this with the upcoming 1.2.0 TestFlight release.
It's currently still waiting for review but should release fairly soon.

**Comment by kgraefe** [comment]
Wow that was fast 👍

**Comment by kgraefe** [comment]
So I tested that for a while and encountered two bugs:
- when you listen to a book A that's part of a series, than listen to another book and go back to the unfinished book A via the "continue" section (German "Weiterhören...") and listen to the end, it will not switch to the next book of the same series
- when you select the sleep timer "end of the chapter" while listening to the last chapter of a book, it would still continue with the next book in the series

Edit: This is on 1.3.0 release.

**Comment by LeoKlaus** [comment]
Thank you for your feedback!

> * when you listen to a book A that's part of a series, than listen to another book and go back to the unfinished book A via the "continue" section (German "Weiterhören...") and listen to the end, it will not switch to the next book of the same series

This is not necessarily a bug, items are only added to the queue automatically if you start them from either a playlist or a series. I see how this isn't intuitive, though.

> * when you select the sleep timer "end of the chapter" while listening to the last chapter of a book, it would still continue with the next book in the series

I've tried reproducing this on my device but the sleep timer seems to work as intended. (For other reasons, the end of chapter sleep timer actually stops about half a second before the actual end of the chapter, so the next file should never be loaded).
Does this happen with every book in your library or just a few?

**Comment by kgraefe** [comment]
> I've tried reproducing this on my device but the sleep timer seems to work as intended. (For other reasons, the end of chapter sleep timer actually stops about half a second before the actual end of the chapter, so the next file should never be loaded). Does this happen with every book in your library or just a few?

I cannot reproduce as well. Maybe it was a user error. 😅

**Comment by LeoKlaus** [comment]
Thank you for the update! You might have just found some edge-case I didn't think of. Let me know if it ever happens again.

I'm still thinking about a good solution to add items to the queue when selecting them from the "Weiterhören" section.
The issue I'm having is that there could be multiple different queues a book could be part of (i.e. a book can be part of multiple collections, series or playlists) and there's no hierarchy between those.

The only solution I can come up with is saving the queue with the playback state, but I'll have to check how that impacts the storage footprint (when using iCloud sync with Jellyfin/Emby, the available storage is extremely limited).

**Comment by GenericRyan** [comment]
Would it be possible Leo to have an option in settings to turn off automatically play next in series? Possibly with an explanation that this setting only applies if you play from the series list. Both my partner and I ran into this feature by accident and were confused as to why the next book was added to the queue. Both of us use the series list to select the next book rather than searching for it, but with only the intention of listening to that one book, not the whole series. I feel that for new users having the next book added to the queue, but only if played from the series list isn't obvious. Hope this makes sense. If I've missed a setting or misunderstood something, let me know. Thank you for your work on this app.

**Comment by kgraefe** [comment]
Q: Where do I find the queue? 🫣

Instead of an option I would scratch the "automatically" from the issue title. Having a "play series from here" entry in the long-press menu of the episode would suit my use case as well. (and having that in the "continue" section would solve the other issue I currently have with the feature)

**Comment by GenericRyan** [comment]
> Q: Where do I find the queue? 🫣

I don't think there is a way to actually see the queue, sorry if I unintentionally gave that impression.

**Comment by LeoKlaus** [comment]
I agree that the automatic (and inconsistent at that) queueing is unintuitive, adding a button to add an entire series to the queue via the context menu is a good idea.

You should see all enqueued items below the chapters in the now playing view:
![image](https://github.com/user-attachments/assets/0d0367be-708a-42ea-9a16-96b7addf15c7)

**Comment by GenericRyan** [comment]
> I agree that the automatic (and inconsistent at that) queueing is unintuitive, adding a button to add an entire series to the queue via the context menu is a good idea.
> 
> You should see all enqueued items below the chapters in the now playing view: 

Ah I hadn't seen that at the bottom, thanks for that, useful to know.

**Comment by kgraefe** [comment]
> adding a button to add an entire series to the queue via the context menu is a good idea.

sorry for nit picking, but adding the **entire series** doesn't help me too much. What I want is the **rest of the series**  starting from the selected episode.
